### PR TITLE
Eng 6729 ReactNative start stop support

### PR DESCRIPTION
**Pull request by kishisaka-nid**
Add ReactNative start / stop support. Tested on iOS (iphone15 simulator) and Android (pixel 4a)

output from metro iOS:

 ```
 Android metro logs
 
 LOG  Running "neuroidReactNativeSDKSandbox" with {"rootTag":11}
 INFO  (NeuroID Info)  ["SiteID set ", "form_misty559"]
 INFO  (NeuroID Info)  ["**** NOTE: THIS METHOD IS DEPRECATED"]
 INFO  (NeuroID Info)  ["**** NOTE: THIS METHOD IS DEPRECATED"]
 DEBUG  (NeuroID Debug)  ["Environment Being Set - Production"]
 DEBUG  (NeuroID Debug)  ["NeuroID Started"]
 INFO  (NeuroID Info)  ["Client ID:", "f9a1bbee-408f-4656-a336-6b371aeda7cd"]
 DEBUG  (NeuroID Debug)  ["setScreenName()", "screenName3131"]
 INFO  (NeuroID Info)  ["Setting User ID: ", "Bfbfbd f_reactnative_sandbox"]
 DEBUG  (NeuroID Debug)  ["startSession(): startSession45235 true"]
 INFO  (NeuroID Info)  ["Setting Registered User ID: ", "registeredUserId4324"]
 DEBUG  (NeuroID Debug)  ["pauseCollection()"]
 DEBUG  (NeuroID Debug)  ["resumeCollection()"]
 DEBUG  (NeuroID Debug)  ["stopSession(): true"]
 DEBUG  (NeuroID Debug)  ["setScreenName()", "Personal Information"]
 
 ----------------
 
 iOS metro logs: 
 
 LOG  Running "neuroidReactNativeSDKSandbox" with {"rootTag":1,"initialProps":{}}
 INFO  (NeuroID Info)  ["SiteID set ", "form_misty559"]
 INFO  (NeuroID Info)  ["**** NOTE: THIS METHOD IS DEPRECATED"]
 INFO  (NeuroID Info)  ["**** NOTE: THIS METHOD IS DEPRECATED"]
 DEBUG  (NeuroID Debug)  ["Environment Being Set - Production"]
 DEBUG  (NeuroID Debug)  ["NeuroID Started"]
 INFO  (NeuroID Info)  ["Client ID:", "4A098D0A-2B2B-4370-8CA6-28F6B08D9D3E"]
 DEBUG  (NeuroID Debug)  ["setScreenName()", "screenName3131"]
 INFO  (NeuroID Info)  ["Setting User ID: ", "Gothkllkkhl_reactnative_sandbox"]
 DEBUG  (NeuroID Debug)  ["startSession(): startSession45235 true"]
 INFO  (NeuroID Info)  ["Setting Registered User ID: ", "registeredUserId4324"]
 DEBUG  (NeuroID Debug)  ["pauseCollection()"]
 DEBUG  (NeuroID Debug)  ["resumeCollection()"]
 DEBUG  (NeuroID Debug)  ["stopSession(): true"]
 DEBUG  (NeuroID Debug)  ["setScreenName()", "Personal Information"]
 DEBUG  (NeuroID Debug)  ["setScreenName()", "Personal Information"]
```

### CODE REVIEW

**Review by quacke-nid — APPROVED** [review]

File: android/src/main/java/com/neuroidreactnativesdk/NeuroidReactnativeSdkModule.kt [hunk 1]
```diff
@@ -1,10 +1,14 @@
 package com.neuroidreactnativesdk
 
 import android.app.Application
+import androidx.annotation.RequiresPermission.Read
```
Comment (kevinsites) [inline android/src/main/java/com/neuroidreactnativesdk/NeuroidReactnativeSdkModule.kt:4]: just curious, what is this import used for?

File: android/src/main/java/com/neuroidreactnativesdk/NeuroidReactnativeSdkModule.kt [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
 import com.facebook.react.bridge.*
 import com.facebook.react.bridge.ReadableMap
 import com.neuroid.tracker.NeuroID
 import com.neuroid.tracker.extensions.setVerifyIntegrationHealth
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.bridge.Arguments
+import com.neuroid.tracker.models.SessionStartResult
```
Comment (kevinsites) [inline android/src/main/java/com/neuroidreactnativesdk/NeuroidReactnativeSdkModule.kt:11]: Are we using this import?

File: ios/NeuroidReactnativeSdk.swift [hunk 3]
```diff
@@ -60,7 +60,7 @@ class NeuroidReactnativeSdk: NSObject {
 
     @objc(setScreenName:withResolver:withRejecter:)
     func setScreenName(screenName: String, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
-        try? NeuroID.setScreenName(screen: screenName)
+        try? NeuroID.setScreenName(screenName)
```
Comment (kevinsites) [inline ios/NeuroidReactnativeSdk.swift:63]: I think we can refactor this to the async/await Bool value instead of a try

**Review by kevinsites — APPROVED** [review]

File: android/src/main/java/com/neuroidreactnativesdk/NeuroidReactnativeSdkModule.kt (hunk 1 above)
Comment (kishisaka-nid) [reply android/src/main/java/com/neuroidreactnativesdk/NeuroidReactnativeSdkModule.kt:4]: I am not sure if it is used. will remove.

File: android/src/main/java/com/neuroidreactnativesdk/NeuroidReactnativeSdkModule.kt (hunk 2 above)
Comment (kishisaka-nid) [reply android/src/main/java/com/neuroidreactnativesdk/NeuroidReactnativeSdkModule.kt]: import cleanup done

File: ios/NeuroidReactnativeSdk.swift (hunk 3 above)
Comment (kishisaka-nid) [reply ios/NeuroidReactnativeSdk.swift:63]: cleaner to leave this be I think.

File: android/src/main/java/com/neuroidreactnativesdk/NeuroidReactnativeSdkModule.kt (hunk 2 above)
Comment (kishisaka-nid) [reply android/src/main/java/com/neuroidreactnativesdk/NeuroidReactnativeSdkModule.kt]: Arguments left, we use this one.

**Review by priya-xavier — DISMISSED** [review]

File: ios/NeuroidReactnativeSdk.swift (hunk 3 above)
Comment (kishisaka-nid) [reply ios/NeuroidReactnativeSdk.swift]: fixed. Thanks!

**Review by kevinsites — APPROVED** [review]

**Review by priya-xavier — APPROVED** [review]

**Pull request merged**